modules/exploits/freebsd/http/junos_phprc_auto_prepend_file.rb

```diff
@@ -0,0 +1,85 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Junos OS PHPRC Environment Variable Manipulation RCE',
+        'Description' => %q{
+          This module exploits a PHP environment variable manipulation vulnerability affecting Juniper SRX firewalls
+          and EX switches. The affected Juniper devices run FreeBSD and every FreeBSD process can access their stdin
+          by opening /dev/fd/0. The exploit also makes use of two useful PHP features. The first being
+          'auto_prepend_file' which causes the provided file to be added using the 'require' function. The second PHP
+          function is 'allow_url_include' which allows the use of URL-aware fopen wrappers. By enabling
+          allow_url_include, the exploit can use any protocol wrapper with auto_prepend_file. The module then uses
+          data:// to provide a file inline which includes the base64 encoded PHP payload.
+        },
+        'Author' => [
+          'Jacob Baines', # Analysis
+          'jheysel-r7',   # Msf module
+          'Ron Bowes'     # Target setup instructions
+        ],
+        'References' => [
+          [ 'URL', 'https://vulncheck.com/blog/juniper-cve-2023-36845'],
```

Comment:
I'd cite watchtowr as well, as they first pointed out the environment variable manipulation: https://labs.watchtowr.com/cve-2023-36844-and-friends-rce-in-juniper-firewalls/